### ISSUE TITLE
IDE0072 Populate switch false positive and code switch generates invalid code

### ISSUE THREAD

**Issue by JakeYallop** [dotnet/roslyn#60201]
**Version Used**: 
```
#error version
Compiler version: '4.1.0-5.22109.6 (0c82c411)'. Language version: preview.
```

**Steps to Reproduce**:
```csharp
//console app top-level program
Console.WriteLine("Hello, World!");

bool? value = false;
//Sometimes we get a warning here but not always, its weird
var a = value switch
{
    null => "s",
    false => "aa",
    true => "aa"
};

namespace SwitchTest
{
    public class Test
    {
        public bool? Value { get; set; }
        public override string ToString()
        {
            //we always get warnings here
            var a = Value switch
            {
                null => nameof(Value),
                false => "aa",
                true => "aa"
            };
            return a;
        }

        public string S
        {
            get
            {
                //and here
                return Value switch
                {
                    null => nameof(Value),
                    false => "aa",
                    true => "aa"
                };
            }
        }
    }
}
```
```editorconfig
# .editorconfig
dotnet_diagnostic.IDE0072.severity = warning
```

Repro here: https://github.com/JakeYallop/SwitchFalsePositiveRepro

**Expected Behavior**:
No warning or code fix shown.

**Actual Behavior**:
Warning shown, and then the subsequent code fix generates incorrect code:

![image](https://user-images.githubusercontent.com/30874283/158676540-4277433a-1af5-4143-b9a7-07b6493e4e79.png)

![image](https://user-images.githubusercontent.com/30874283/158675765-f3585cf3-78a9-4f1c-8f4d-cfdd87fb9f07.png)

**Other Information**
Visual Studio Version: 17.1.1

**Comment by Youssef1313** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/48876
